sdk/ml/azure-ai-ml/azure/ai/ml/_internal/_schema/component.py

```diff
@@ -118,17 +119,26 @@ def _serialize(self, obj, *, many: bool = False):
     def add_param_overrides(self, data, **kwargs):
         source_path = self.context.pop(SOURCE_PATH_CONTEXT_KEY, None)
         if isinstance(data, dict) and source_path and os.path.isfile(source_path):
+            def should_node_overwritten(_root, _parts):
+                parts = _parts.copy()
+                parts.pop()
+                parts.append("type")
+                _input_type = pydash.get(_root, parts, None)
+                return isinstance(_input_type, str) and _input_type.lower() not in ["boolean"]
```

Comment:
@brynn-code I think it would be better if we can address the unexpected 1 => 1.0 issue. But this PR can be a temporary solution to make anonymous component reuse work on main.